analysisTool.ts

```diff
@@ -191,7 +191,7 @@ export class AnalysisTool {
             (filename: string, data: string) => this.checkFileForScriptingLanguage(filename, data),
             (filename: string, data: string) => this.checkFileForComponent(filename, data),
         ];
-        if (currentPath.substr(-3) === '.js' || this.fileHasTsExtension(currentPath) || currentPath.substr(-5) === '.html' || currentPath.substr(-5) === '.json') {
+        if ((currentPath.substr(-3) === '.js' || this.fileHasTsExtension(currentPath) || currentPath.substr(-5) === '.html' || currentPath.substr(-5) === '.json') && fs.lstatSync(currentPath).isFile()) {
```

Comment:
It seems like you would want this `isFile()` at a higher level so that it would apply whether the directory had `.json` or `.html` in it, basically it shouldn't check any of these file extensions unless it's dealing with a file.

Reply:
It sounds like an optimization, my solution worked for me allowing me to use the tool, so I decided to share. If you believe there's a better solution - please create a new PR, I don't use this tool anymore and didn't find it to be particularly helpful. Cheers!

Reply:
@Splaktar I wonder if you think it's ok to merge?